htdocs/core/lib/files.lib.php

```diff
@@ -1674,11 +1674,12 @@ function dol_add_file_process($upload_dir, $allowoverwrite = 0, $donotupdatesess
 				}
 
 				// dol_sanitizeFileName the file name and lowercase extension
+				$infoTFile = pathinfo($upload_dir."/".$TFile['name'][$i]);
 				$info = pathinfo($destfull);
-				$destfull = $info['dirname'].'/'.dol_sanitizeFileName($info['filename'].($info['extension'] != '' ? ('.'.strtolower($info['extension'])) : ''));
+				$destfull = $info['dirname'].'/'.dol_sanitizeFileName($info['filename'].(!empty($info['extension']) ? ('.'.strtolower($info['extension'])) : (!empty($infoTFile['extension']) ? '.'.strtolower($infoTFile['extension']) : '')));
```

Comment:
Why adding a backfall on infoTFile['extension'] ?
Do you have an exemple of value we have for file we upload and the result we get on $destfull for this case that need the $infoTFile ?